pytest_splunk_addon/splunk.py

```diff
@@ -540,52 +619,234 @@ def uf_external(request):
-        root_tmp_dir = tmp_path_factory.getbasetemp().parent
-        fn = root_tmp_dir / "pytest_docker"
-        with FileLock(str(fn) + ".lock"):
-            docker_services.start("splunk")
+    os.environ["TEST_RUNNER_DIRECTORY"] = os.getcwd()
+    parser = conf_parser.TABConfigParser()
+    parser.read(
+        os.path.join(
+            "{0}".format(os.getenv("SPLUNK_APP_PACKAGE")), "default", "app.conf"
+        )
+    )
+    splunk_addon_name = parser.get("package", "id")
+    try:
+        spl_path = os.path.join(
+            "{0}".format(os.getenv("TEST_RUNNER_DIRECTORY")), "tests", "src"
+        )
+        splunk_spl_name = [
+            filename
+            for filename in os.listdir(spl_path)
+            if filename.startswith(("{0}-").format(splunk_addon_name))
+        ]
+        SPLUNK_ADDON = str(splunk_spl_name[0])
+        LOGGER.info(SPLUNK_ADDON)
+    except Exception as e:
+        LOGGER.error(
+            "ADDON SPL doesn't exist for {0} at tests/src".format(splunk_addon_name)
+        )
+    NAMESPACE_NAME = str(splunk_addon_name.replace("_", "-").lower())
+    LOGGER.info("NAMESPACE_NAME is {}".format(NAMESPACE_NAME))
+    os.environ["NAMESPACE_NAME"] = NAMESPACE_NAME
+    os.environ["SPLUNK_ADDON"] = SPLUNK_ADDON
+    if (
+        "PYTEST_XDIST_WORKER" not in os.environ
+        or os.environ.get("PYTEST_XDIST_WORKER") == "gw0"
+    ):
+        LOGGER.info(
+            "Starting kubernetes_service=splunk for worker id {}".format(
+                str(os.environ.get("PYTEST_XDIST_WORKER"))
+            )
+        )
+        LOGGER.info(
+            "TEST_RUNNER_DIRECTORY : {}".format(
+                str(os.environ.get("TEST_RUNNER_DIRECTORY"))
+            )
+        )
+        current_path = os.path.join(
+            os.path.dirname(os.path.abspath(__file__)),
+            "k8s_manifests",
+            "splunk_standalone",
+        )
+        pattern = r"https:\/\/github\.com\/splunk\/splunk-configurations-base-indexes\/releases\/download\/v[0-9.]+\/splunk_configurations_base_indexes-[0-9.]+.spl"
+        try:
+            response = requests.get(
+                "https://api.github.com/repos/splunk/splunk-configurations-base-indexes/releases/latest"
+            )
+            regex_match = re.search(pattern, str(response.json()))
+            if regex_match:
+                os.environ["SC4S_INDEX_URL"] = regex_match.group()
+        except Exception as e:
+            LOGGER.error(
+                "Exception occurred while fetching SC4S_INDEX_URL : {0}".format(e)
+            )
+        print(os.getenv("SC4S_INDEX_URL"))
+        update_k8s_manifest_files(folder=current_path, file="namespace")
+        update_k8s_manifest_files(folder=current_path, file="splunk_standalone")
+        LOGGER.info("Setting up Splunk")
+        kubernetes_helper_splunk = KubernetesHelper()
+        LOGGER.info(current_path)
+        LOGGER.info(current_path + "/namespace_updated.yaml")
```

Comment:
Can you please clean up those log messages and some print statements in this PR if they are not needed?

Reply:
Yes, removed unnecessary logger statements and print statements.